sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2009,8 +2010,15 @@ def provision_with_retries(
                 else:
                     cloud_user = to_provision.cloud.get_current_user_identity()
                 # Skip if to_provision.cloud does not support requested features
-                to_provision.cloud.check_features_are_supported(
-                    self._requested_features)
+                granted_features = to_provision.cloud.grant_features(
+                    self._requested_features,
+                    ExcludableFeatureCheckConfig(cluster_name=cluster_name))
+
+                if self._requested_features != granted_features:
+                    logger.info(
+                        f'{colorama.Fore.CYAN}The following features will be skipped since they are not supported by {to_provision.cloud.__class__.__name__} and were deemed optional : '
```

Comment:
```suggestion
                        f'{colorama.Fore.CYAN}The following features will be skipped since they are not supported by {to_provision.cloud._REPR} and were deemed optional : '
```

Reply:
Checked and `print(to_provision.cloud)` gives `Kubernetes` so just `{to_provision.cloud}` should be fine

Reply:
IIUC using `cloud._REPR` is some code convention. cc @concretevitamin for a look

Reply:
Yeah, it's supposed to be used either via `{to_provision.cloud}` or `{repr(to_provision.cloud)}` to not access private attrs directly.

Reply:
Changed to `{to_provision.cloud}`